### ISSUE TITLE
Faulty core fixes: Nested `foreach` incorrectly indented.

### ISSUE THREAD

**Issue by pento** [WordPress-Coding-Standards/WordPress-Coding-Standards#1121]
Given the following example:

```php
<?php

foreach ( $a as $b ) :
?>
<?php
	foreach ( $b as $c ) :
	endforeach;
endforeach;
```

The inner `foreach` is incorrectly losing its indent.

**Comment by jrfnl** [comment]
@pento This is an awkward one. I'm actually tempted to class it as "garbage in, garbage out"...

Both the scope indent sniff as well as the embedded PHP sniff have some code handling the indents here.
The ones from the embeddedPHP sniff have been disabled in the WP rulesets, but would make no significant difference here anyhow.

The problem is one of reverse-logic.

For scope indents to be determined, something is needed as a starting point. The scope indent sniff uses PHP open tags as such and resets expected indent each time one is encountered.

As the input has the PHP open tag on a line with an incorrect indent, the rest of the code is influenced by that and will be incorrectly indented too.

If the input would have been the below, it would have been fine:

```php
foreach ( $a as $b ) :
	?>
	<?php
	foreach ( $b as $c ) :
	endforeach;
endforeach;
```

I very much doubt this can be fixed by patching a sniff.

P.S.: I'm presuming the subsequent closing and opening PHP tags are only for the example code ? If not, they should be removed completely.

**Comment by pento** [comment]
Yeah, the example code was the most cut down version I could make.

I don't making this `wontfix` if it's too hard to address, there are only a [handful of instances](https://github.com/WordPress/wordpress-develop/search?utf8=%E2%9C%93&q=endforeach+path%3Asrc&type=) of `endforeach` in Core.

**Comment by jrfnl** [comment]
> there are only a handful of instances of endforeach in Core.

It's not necessarily the `endforeach` which is the problem here, but the incorrectly placed PHP open tag just before it.

I kind of think we'll probably need a handful of manual fixes prior to running the fixer anyway, so I'd suggest these ones qualify for that.

**Comment by pento** [comment]
👍🏻
